src/transformer.ts

```diff
@@ -2,12 +2,27 @@ import Tag from './tag';
 import { Class } from './schema-types/class';
 import { isPromise } from './utils';
 import type { Config, Node, NodeType, Schema, Transformer } from './types';
+import { Value } from '..';
 
 type AttributesSchema = Schema['attributes'];
 
 export const globalAttributes: AttributesSchema = {
   class: { type: Class, render: true },
-  id: { type: String, render: true },
+  id: {
+    type: String,
```

Comment:
```suggestion
    type: ID,
```

And add this to `src/schema-types/class`:

```ts
export class ID implements CustomAttributeTypeInterface {
  validate(value: any): ValidationError[] {
    if (typeof value === 'string' && value.match(/^[a-zA-Z]/)) return [];

    return [
      {
        id: 'attribute-type-invalid',
        level: 'error',
        message: "The 'id' attribute must start with a letter",
      },
    ];
  }
}
```